extensions/amp-web-push/0.1/web-push-service.js

```diff
@@ -329,14 +329,12 @@ export class WebPushService {
    * @return {Promise<ServiceWorkerRegistrationMessage>}
    */
   registerServiceWorker() {
+
     return this.queryHelperFrame_(
         WindowMessenger.Topics.SERVICE_WORKER_REGISTRATION,
         {
-          workerUrl: this.config_['service-worker-url'],
-          registrationOptions: this.config_.serviceWorkerRegistrationOptions ||
-            {
-              scope: '/',
-            },
+          workerUrl : this.config_['service-worker-url'],
+          registrationOptions : (this.config_['service-worker-scope'] ? {scope : this.config_['service-worker-scope']} : {})
```

Comment:
```suggestion
          registrationOptions : {
            scope: this.config_['service-worker-scope'] || '/',
          },
```

Reply:
I thought about this, but there's an issue:

Setting the scope to "/" declares that the SW has to be at the root. If service-worker-url is, for example: /example/sw.js, and there's no service-worker-scope provided, passing a registrationOptions object with a scope at the root will break the registration.

If we create an empty registrationOptions object (without any scope), when the service-worker-scope is not set, the SW will get register on its current location and will control any subpath under its current directory.

That's why the condition is to declare a scope, only when service-worker-scope is received, otherwise, returning an empty registrationOptions object (since it can't be undefined).

Let me know your thoughts.

Reply:
Hmm, that makes sense.